frontend/integration-tests/views/sidenav.view.ts

```diff
@@ -1,10 +1,16 @@
-import { $$, by, browser, element, ExpectedConditions as until } from 'protractor';
+/* eslint-disable no-undef, no-unused-vars */
 
-export const navSectionFor = (name: string) => element(by.cssContainingText('.pf-c-nav > .pf-c-nav__list > .pf-c-nav__item', name));
+import {$$, by, browser, element, ExpectedConditions as until} from 'protractor';
+
+export const navSectionFor = (name: string) => $(`[data-test-nav-item="${name}"]`);
 
 export const clickNavLink = (path: [string, string]) => browser.wait(until.visibilityOf(navSectionFor(path[0])))
-  .then(() => navSectionFor(path[0]).click())
-  .then(() => browser.wait(until.visibilityOf(navSectionFor(path[0]).element(by.linkText(path[1])))))
-  .then(() => navSectionFor(path[0]).element(by.linkText(path[1])).click());
+  .then(() => navSectionFor(path[1]).isDisplayed().then(async (isVisible: boolean) => {
+    if (!isVisible) {
+      await navSectionFor(path[0]).click();
```

Comment:
You're mixing promises and `async` here. It would be better to convert `clickNavLink` entirely to an `async` function.